File: 02-programacion-basica/11-archivos-y-persistencia-basica/java/practicas/PracticaRegistroArchivo.java
import java.io.IOException;
import java.nio.file.Files;
import java.nio.file.Path;
import java.util.List;

public class PracticaRegistroArchivo {
    public static void main(String[] args) {
        Path archivo = Path.of("registro-cursos.txt");
        List<String> cursos = List.of("Java", "Scala", "Git");

        try {
            // Files.write() guarda cada elemento de la lista como una línea.
            Files.write(archivo, cursos);
            for (String curso : Files.readAllLines(archivo)) {
                System.out.println("Curso: " + curso);
            }
            Files.deleteIfExists(archivo);
        } catch (IOException e) {
            System.out.println("No se pudo acceder al archivo: " + e.getMessage());
        }
    }
}

File: 02-programacion-basica/11-archivos-y-persistencia-basica/scala/practicas/PracticaRegistroArchivo.scala
import java.nio.file.{Files, Path}
import java.nio.charset.StandardCharsets
import scala.jdk.CollectionConverters.*

@main def practicaRegistroArchivo(): Unit =
  val archivo = Path.of("registro-cursos.txt")
  val cursos = List("Java", "Scala", "Git")
  Files.write(archivo, cursos.asJava, StandardCharsets.UTF_8)
  // asScala adapta el resultado de la API Java para recorrerlo como colección Scala.
  Files.readAllLines(archivo, StandardCharsets.UTF_8).asScala.foreach(c => println(s"Curso: $c"))
  Files.deleteIfExists(archivo)
